dbeaver/dbeaver#21574 eclipse configuration directory in your profile

serjiokov opened the pull request:
The configuration options specified in order to store data in OS specific folders.

serjiokov committed:
dbeaver/dbeaver#21574 DBeaver creates a ".eclipse/configuration"

Provided product setting for store configuration in OS specific folder.

serjiokov committed:
dbeaver/dbeaver#21574 DBeaver creates a ".eclipse/configuration"

Extra change reverted.

serge-rider commented on product/community/DBeaver.product:
```diff
@@ -94,6 +94,9 @@
         <plugin id="org.eclipse.equinox.event" autoStart="true" startLevel="2"/>
         <!-- This is important because simpleconfigurator conflicts with update.configurator -->
         <plugin id="org.eclipse.update.configurator" autoStart="false" startLevel="10"/>
```
conf -> "install-data"

serjiokov replied:
done

serge-rider commented on product/community/DBeaver.product:
```diff
@@ -94,6 +94,9 @@
         <!-- This is important because simpleconfigurator conflicts with update.configurator -->
         <plugin id="org.eclipse.update.configurator" autoStart="false" startLevel="10"/>
+        <property name="osgi.configuration.area" value="@user.home/AppData/Roaming/DBeaverData/conf" os="win32" />
+        <property name="osgi.configuration.area" value="@user.home/.config/DBeaverData/conf" os="linux" />
```
Not sure about Linux path. It has to be in $XDG_DATA_HOME

serjiokov replied:
on OS the $XDX_DATA_HOME have to be resolved into 

![image](https://github.com/dbeaver/dbeaver/assets/32031248/f3dae197-bc9b-48eb-951e-a274ee846214)

serjiokov committed:
dbeaver/dbeaver#21574 DBeaver creates a ".eclipse/configuration"

Renamed conf to intall-data

serjiokov committed:
dbeaver/dbeaver#21574 Product creates a ".eclipse/configuration"
directory in your profile for each version

The new dbeaver launcher provided for customization configuration area
location

serjiokov committed:
dbeaver/dbeaver#21574 Product creates a ".eclipse/configuration"
directory in your profile for each version

Rewraped product file

serjiokov commented:
For now reached next points:
1. Organized own dbeaver launcher with implemetation of store relative configurtaion (../DBeaverData/install-data) in OS specific location:
    a. WIN - APPDATA
    b. UNIX - XDG_DATA_HOME 
In order to achive it next properties in product require:
        <property name="osgi.configuration.area" value="**@data.home**/DBeaverData/install-data" os="win32" />
        <property name="osgi.configuration.area" value="**@data.home**/DBeaverData/install-data" os="linux" />
        <property name="osgi.configuration.area" value="**@data.home**/DBeaverData/install-data" os="macosx" />

Not achived:
Generation .ini file option (or phaze of product materialization) still paste 
"-startup 
  plugins/org.eclipse.equinox.launcher_1.6.500.v20230717-2134.jar" that does not fit requiremnts.
This line have to be changed to 
" -startup
plugins/org.jkiss.dbeaver.launcher_[version]jar" 

Result:
Product configuration data will be stored at `"~/AppData/DBeaverData/install-data"`
The "~/.eclipse" not used any more.

Commit:
Merge branch 'devel' into dbeaver/dbeaver#21574-eclipse-configuration

Commit:
dbeaver/dbeaver#21574 Eclipse configuration launcher

Commit:
Merge branch 'dbeaver/dbeaver#21574-eclipse-configuration' of github.com:dbeaver/dbeaver into dbeaver/dbeaver#21574-eclipse-configuration

Commit:
beaver/dbeaver#21574 Specify global monitor for Entity Diagram
association processing

Bundle version fixed

Commit:
beaver/dbeaver#21574 Specify global monitor for Entity Diagram
association processing

feature.xml formated

Commit:
beaver/dbeaver#21574 Specify global monitor for Entity Diagram
association processing

feature.xml formated

Commit:
beaver/dbeaver#21574 Specify global monitor for Entity Diagram
association processing

Fixed bundle version

Commit:
beaver/dbeaver#21574 Specify global monitor for Entity Diagram
association processing

Fixed bundle build

Commit:
beaver/dbeaver#21574 Specify global monitor for Entity Diagram
association processing

Fixed bundle build

Commit:
dbeaver/dbeaver#21574 eclipse configuration directory in your profile

Moved feature to standalone feature

Commit:
dbeaver/dbeaver#21574 eclipse configuration directory in your profile

Fixed version of launcher bundle

Commit:
dbeaver/dbeaver#21574 eclipse configuration directory in your profile

Fixed initial location for launcher

Commit:
dbeaver/dbeaver#21574 Fixed initial location for launcher

Fixed initial location for launcher

serjiokov committed:
dbeaver/dbeaver#21574 Fixed product file for launcher

Commit:
dbeaver/dbeaver#21574 Reformatted code

Commit:
Merge remote-tracking branch 'origin/dbeaver/dbeaver#21574-eclipse-configuration' into dbeaver/dbeaver#21574-eclipse-configuration

Commit:
Merge branch 'devel' into dbeaver/dbeaver#21574-eclipse-configuration

Commit:
dbeaver/dbeaver#21574 Removed launcher from product file

Commit:
dbeaver/dbeaver#21574 Reformatted code

Commit:
dbeaver/dbeaver#21574 Product file code style

Commit:
dbeaver/dbeaver#21574 Product file code style

serge-rider commented on plugins/org.jkiss.dbeaver.launcher/src/org/eclipse/equinox/launcher/Constants.java:
```diff
@@ -0,0 +1,149 @@
+/*
```
Can we just import all this stuff from original launcher and not copy-paste it?
We have it in plugin list anyway.

serjiokov replied:
I don't think so we targeted to reimplement a private method 
` private URL buildLocation(String property, URL defaultLocation, String userDefaultAppendage) `

serge-rider replied:
Yes. But why can't we just import constants, JNI bridge and other stuff which is simply copy-pasted?

serjiokov replied:
Done, but one thing the equinox classes is in internal package.

serjiokov committed:
dbeaver/dbeaver#21574-eclipse-configuration

Imported classes from original equinox bundle

serjiokov committed:
Merge branch 'dbeaver/dbeaver#21574-eclipse-configuration' of
git@github.com:dbeaver/dbeaver.git into
dbeaver/dbeaver#21574-eclipse-configuration

Matvey16 committed:
Merge branch 'devel' into dbeaver/dbeaver#21574-eclipse-configuration

serjiokov committed:
Merge branch 'devel' into dbeaver/dbeaver#21574-eclipse-configuration

serjiokov committed:
dbeaver/dbeaver#21574-eclipse-configuration

Fixed package private dependency

serjiokov committed:
Merge branch 'dbeaver/dbeaver#21574-eclipse-configuration' of github.com:dbeaver/dbeaver into dbeaver/dbeaver#21574-eclipse-configuration

serjiokov committed:
Merge branch 'dbeaver/dbeaver#21574-eclipse-configuration' of
github.com:dbeaver/dbeaver into
dbeaver/dbeaver#21574-eclipse-configuration

serjiokov committed:
dbeaver/dbeaver#21574 eclipse configuration directory in your profile

Fixed manifest version identification

Matvey16 committed:
Merge branch 'devel' into dbeaver/dbeaver#21574-eclipse-configuration

Commit:
dbeaver/dbeaver#21574 eclipse configuration directory in your profile

Added resolve ~ path
Renamed package to dbeaver

Commit:
Merge branch 'dbeaver/dbeaver#21574-eclipse-configuration' of github.com:dbeaver/dbeaver into dbeaver/dbeaver#21574-eclipse-configuration

Commit:
dbeaver/dbeaver#21574-eclipse-configuration

Reorganized package to original org.eclipse.equinox.launcher

Commit:
dbeaver/dbeaver#21574-eclipse-configuration

Read version from .eclipseproduct

Commit:
dbeaver/dbeaver#21574-eclipse-configuration

Read version from .eclipseproduct (part -2)

Commit:
dbeaver/dbeaver#21574-eclipse-configuration

Added Api for retrieving version from .eclipse

Commit:
dbeaver/dbeaver#21574-eclipse-configuration

Removed class JNIBridge from local package

Commit:
dbeaver/dbeaver#21574-eclipse-configuration

Removed debug flag

Commit:
dbeaver/dbeaver#21574-eclipse-configuration

Returned JNIBridge

Matvey16 committed:
Merge branch 'devel' into dbeaver/dbeaver#21574-eclipse-configuration

Matvey16 committed:
Merge branch 'devel' into dbeaver/dbeaver#21574-eclipse-configuration

Matvey16 commented:
Verified

Commit:
Merge branch 'dbeaver/pro#2197-encrypted-project' of
git@github.com:dbeaver/dbeaver.git into
dbeaver/pro#2197-encrypted-project

Commit:
Merge branch 'dbeaver/dbeaver#21574-eclipse-configuration' of github.com:dbeaver/dbeaver into dbeaver/dbeaver#21574-eclipse-configuration

Commit:
dbeaver/dbeaver#21574-eclipse-configuration

Empty line removed

Commit:
Merge branch 'dbeaver/dbeaver#21574-eclipse-configuration' of
github.com:dbeaver/dbeaver into
dbeaver/dbeaver#21574-eclipse-configuration

Matvey16 commented:
Verified
